vpr/src/place/place.cpp

```diff
@@ -356,11 +443,26 @@ static void update_placement_cost_normalization_factors(t_placer_costs* costs, c
 
 static double get_total_cost(t_placer_costs* costs, const t_placer_opts& placer_opts, const t_noc_opts& noc_opts);
 
-static double get_net_cost(ClusterNetId net_id, t_bb* bb_ptr);
+static double get_net_cost(ClusterNetId net_id, const t_bb& bbptr);
+
+static double get_net_layer_cost(ClusterNetId /* net_id */,
+                                 const std::vector<t_2D_bb>& bbptr,
+                                 const std::vector<int>& layer_pin_sink_count);
+
+static void get_bb_from_scratch(ClusterNetId net_id,
+                                t_bb& coords,
+                                t_bb& num_on_edges);
 
-static void get_bb_from_scratch(ClusterNetId net_id, t_bb* coords, t_bb* num_on_edges);
+static void get_layer_bb_from_scratch(ClusterNetId net_id,
+                                      std::vector<t_2D_bb>& num_on_edges,
+                                      std::vector<t_2D_bb>& coords,
+                                      std::vector<int>& layer_pin_sink_count);
 
-static double get_net_wirelength_estimate(ClusterNetId net_id, t_bb* bbptr);
+static double get_net_wirelength_estimate(ClusterNetId net_id, const t_bb& bbptr);
```

Comment:
`bbptr` doesn't look like a ptr/output reference to me, so probably better to call it `bb`. Applies to all cases of `bbptr`, even when it's a ptr, since that would be visible from the type/usage.

Reply:
Done in the PR that I refactored the place.cpp file